persistence/persistence-api/src/main/java/org/cbioportal/persistence/util/CustomEhCachingProvider.java

```diff
@@ -40,43 +43,104 @@
 import org.ehcache.config.builders.ResourcePoolsBuilder;
 import org.ehcache.config.Configuration;
 import org.ehcache.config.units.MemoryUnit;
+import org.ehcache.xml.XmlConfiguration;
 import org.ehcache.core.config.DefaultConfiguration;
 import org.ehcache.jsr107.EhcacheCachingProvider;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Value;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.ehcache.impl.config.persistence.DefaultPersistenceConfiguration;
+import org.cbioportal.persistence.CacheEnabledConfig;
 
 public class CustomEhCachingProvider extends EhcacheCachingProvider {
 
     private static final Logger LOG = LoggerFactory.getLogger(CustomEhCachingProvider.class);
 
     @Value("${ehcache.xml_configuration:/ehcache.xml}")
-    private String xmlConfiguration;
+    private String xmlConfigurationFile;
 
-    @Value("${ehcache.cache_enabled:false}")
-    private Boolean cacheEnabled;
+    @Value("${ehcache.cache_type:none}")
+    private String cacheType;
+
+    @Value("${ehcache.general_repository_cache.max_mega_bytes_heap:1024}")
+    private Integer generalRepositoryCacheMaxMegaBytes;
+
+    @Value("${ehcache.static_repository_cache_one.max_mega_bytes_heap:30}")
+    private Integer staticRepositoryCacheOneMaxMegaBytes;
+
+    @Value("${ehcache.persistence_path:/tmp/}")
+    private String persistencePath;
+
+    @Value("${ehcache.general_repository_cache.max_mega_bytes_local_disk:4096}")
+    private Integer generalRepositoryCacheMaxMegaBytesLocalDisk;
+
+    @Value("${ehcache.static_repository_cache_one.max_mega_bytes_local_disk:32}")
+    private Integer staticRepositoryCacheOneMaxMegaBytesLocalDisk;
 
     @Override
     public CacheManager getCacheManager() {
 
         CacheManager toReturn = null;
-         try {
-            if (cacheEnabled != null && cacheEnabled) {
-                LOG.info("Caching is enabled, using '" + xmlConfiguration + "' for configuration");
-                toReturn = this.getCacheManager(getClass().getResource(xmlConfiguration).toURI(),
-                                            getClass().getClassLoader());
+        try {
+            if (CacheEnabledConfig.enableCache(cacheType)) {
+                LOG.info("Caching is enabled, using '" + xmlConfigurationFile + "' for configuration");
+                XmlConfiguration xmlConfiguration = new XmlConfiguration(getClass().getResource(xmlConfigurationFile));
```

Comment:
it might be a good idea to catch the case where getResource(xmlConfigurationFile) returns null .. that way we can do something like LOG.error("attempt to load Ehcache xml configuration file " + xmlConfiguration + " failed -- file is not available in the classpath") rather than an NPE